server/src/test/java/org/opensearch/index/translog/transfer/TranslogTransferManagerTests.java

```diff
@@ -186,66 +191,92 @@ public void testReadMetadataNoFile() throws IOException {
             remoteBaseTransferPath,
             null
         );
+        doAnswer(invocation -> {
+            LatchedActionListener<List<BlobMetadata>> latchedActionListener = invocation.getArgument(2);
+            List<BlobMetadata> bmList = new LinkedList<>();
+            latchedActionListener.onResponse(bmList);
+            return null;
+        }).when(transferService).listAllInSortedOrder(any(BlobPath.class), anyInt(), any(ActionListener.class));
 
-        when(transferService.listAll(remoteBaseTransferPath)).thenReturn(Sets.newHashSet());
         assertNull(translogTransferManager.readMetadata());
     }
 
-    public void testReadMetadataSingleFile() throws IOException {
+    public void testReadMetadataHappy() throws IOException {
```

Comment:
nit - let's rename it to `testReadMetadata()`? (and a java doc saying it's for happy case :) )